superset/db_engine_specs/postgres.py

```diff
@@ -289,9 +289,12 @@ def get_cancel_query_id(cls, cursor: Any, query: Query) -> Optional[str]:
         :param query: Query instance
         :return: Postgres PID
         """
-        cursor.execute("SELECT pg_backend_pid()")
-        row = cursor.fetchone()
-        return row[0]
+        try:
+            cursor.execute("SELECT pg_backend_pid()")
+            row = cursor.fetchone()
+            return row[0]
+        except Exception:
```

Comment:
Minor improvement proposal: is there a more specific exception type that we could be catching here?

Reply:
I've updated the PR comment with the error being raised.  It's a `superset.exceptions.SupersetErrorsException`.  Would you like me to catch that instead of the generic exception?

Reply:
> I've updated the PR comment with the error being raised.  It's a `superset.exceptions.SupersetErrorsException`.  Would you like me to catch that instead of the generic exception?

Really? I would have thought it's a psycopg2 or at least a SQLAlchemy exception, as I assume it's coming from the cursor call. But yeah, let's make it as specific as possible if it's not too much trouble.